### PR TITLE
[quantization] Linear+BatchNorm1d fusion support for QAT

### PR DESCRIPTION
**Pull request by shoonee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66891

Add QAT support for fused Linear+BatchNorm1d, addressing https://github.com/pytorch/pytorch/issues/66483.
Added tests to check module types when fused and module outputs against a reference.

Differential Revision: [D31964610](https://our.internmc.facebook.com/intern/diff/D31964610)

### CODE REVIEW

**Comment by vkuzo** [comment]
looks great!  There are just some build failures because of rebase conflicts, can we rebase onto a newer revision?

**Comment by sean-ngo** [comment]
@sean-ngo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31964610).

**Comment by sean-ngo** [comment]
@sean-ngo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31964610).

**Comment by vkuzo** [comment]
There is a test failure on Mac OS which looks transient, could you rebase again?

**Comment by sean-ngo** [comment]
@sean-ngo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31964610).

**Review by vkuzo — APPROVED** [review]
Thank you for working on this, looks great!

nit: could we rename the PR to mention this is part 1 out of 2, as a future PR will have to add the convert functionality?

**Comment by sean-ngo** [comment]
@sean-ngo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31964610).

**Comment by sean-ngo** [comment]
@sean-ngo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31964610).

**Comment by vkuzo** [comment]
https://github.com/pytorch/pytorch/pull/72431 is a newer version of this
